build/xtask/src/dist.rs

```diff
@@ -431,6 +431,14 @@ pub fn package(
 
         translate_srec_to_other_formats(&cfg, image_name, "combined")?;
 
+        if cfg.toml.target.contains("riscv") {
```

Comment:
The way the build flow happens with these changes is:
1. Build srec image
2. Translate srec to other formats (bin, ihex) for rv and (bin, ihex, elf) for !rv
3. Build elf image for rv

I think we should instead create a consistent flow for all architectures:
1. Build elf image
2. Translate to all other formats (ihex, bin, srec)

That way we won't have to maintain different paths here for different architectures. Is there any limitation currently because of which we cannot do that?

Reply:
The only sticking point I'm aware of is the additional ram-section, which we certainly don't want in binaries aimed elsewhere.

I took this approach from a position of cautious change - in particular, I didn't want to change this for non-rv targets, as their Qemu HW might take a different approach.

I agree in principle, however, that it would be better to have a consistent flow (or a 'true' federated image generation, where the app.toml has a mechanism to specify image generation instructions).

Reply:
All targets will now use the manual ELF generation. There's one more change coming to not break ARM, et al, but the groundwork is in.

Reply:
> The only sticking point I'm aware of is the additional ram-section, which we certainly don't want in binaries aimed elsewhere.

I don't think we need it even for our own builds. Based on @zachreizner's comment, it looks like a hack that was done for a specific case. I see that you decided to drop the hack in the latest commit.